extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/DevUIConfig.java

```diff
@@ -26,6 +26,7 @@ public class DevUIConfig {
      * More hosts allowed for Dev UI
      *
      * Comma separated list of valid URLs, e.g.: www.quarkus.io, myhost.com
+     * (This can also be a regex)
```

Comment:
I think this is a bit brittle. And given it's a security feature, I'm a bit worried about it.

The fact that anything being a valid pattern makes it a pattern means that `my.host.com` will be a Pattern. Meaning that `my2host.com` will match.

I think we need to be a lot more careful about it and take a step back.

I don't know how it's handled in other areas where we have this kind of thing.

/cc @cescoffier

Reply:
I dropped the `triage/backport` label for now. We need to discuss this.

Reply:
Well could reduce this and use just a prefix (with a startsWith and not a regexp)

Reply:
I don't know what we're doing for the CORS for instance? We should probably try to be consistent.

Reply:
@gsmet I first look for a complete match (like we always have done) before I then check for a regex. Regex has to a valid regex (starting with ^)